Jenkinsfile

```diff
@@ -78,39 +78,53 @@ pipeline {
-                    MANIFEST_NAME = "app.json"
+                BUILD_VERSION = "PKG_VERSION"
+                CHANNEL = "stable"
+                SERVICE_NAME = "fdc3"
+                MANIFEST_NAME = "app.json"
 
-                    S3_LOC = env.DSERVICE_S3_ROOT + SERVICE_NAME + "/" + BUILD_VERSION
-                    DOCS_CHANNEL_LOC = env.DSERVICE_S3_ROOT_DOCS + SERVICE_NAME + "/" + CHANNEL
-                    DOCS_VERSIONED_LOC = env.DSERVICE_S3_ROOT_DOCS + SERVICE_NAME + "/" + BUILD_VERSION
-                    MANIFEST_LOC = env.DSERVICE_S3_ROOT + SERVICE_NAME + "/" + MANIFEST_NAME
-                }
-                sh "npm i --ignore-scripts"
-                sh "SERVICE_VERSION=${BUILD_VERSION} npm run build"
-                sh "echo ${GIT_SHORT_SHA} > ./dist/SHA.txt"
-                sh "npm run zip"
-                sh "npm run docs"
-                sh "aws s3 cp ./res/provider ${S3_LOC}/ --recursive"
-                sh "aws s3 cp ./dist/provider ${S3_LOC}/ --recursive"
-                sh "aws s3 cp ./dist/client/openfin-" + SERVICE_NAME + ".js ${S3_LOC}/"
-
-                sh "aws s3 cp ./dist/docs ${DOCS_CHANNEL_LOC} --recursive"
-                sh "aws s3 cp ./dist/docs ${DOCS_VERSIONED_LOC} --recursive"
-                sh "aws s3 cp ./dist/provider/app.json ${MANIFEST_LOC}"
+                S3_LOC = "env.DSERVICE_S3_ROOT" + "SERVICE_NAME" + "/" + "BUILD_VERSION"
+                DOCS_CHANNEL_LOC = "env.DSERVICE_S3_ROOT_DOCS" + "SERVICE_NAME" + "/" + "CHANNEL"
+                DOCS_VERSIONED_LOC = "env.DSERVICE_S3_ROOT_DOCS" + "SERVICE_NAME" + "/" + "BUILD_VERSION"
+                MANIFEST_LOC = "env.DSERVICE_S3_ROOT" + "SERVICE_NAME" + "/" + "MANIFEST_NAME"
+            }
+            stages {
+                stage("Build & Deploy to CDN (Production)") {
+                    steps {
+                        sh "npm i --ignore-scripts"
+                        sh "SERVICE_VERSION=${BUILD_VERSION} npm run build"
+                        sh "echo ${GIT_SHORT_SHA} > ./dist/SHA.txt"
+                        sh "npm run zip"
+                        sh "npm run docs"
+                        sh "aws s3 cp ./res/provider ${S3_LOC}/ --recursive"
+                        sh "aws s3 cp ./dist/provider ${S3_LOC}/ --recursive"
+                        sh "aws s3 cp ./dist/client/openfin-" + SERVICE_NAME + ".js ${S3_LOC}/"
 
-                withCredentials([string(credentialsId: "NPM_TOKEN_WRITE", variable: 'NPM_TOKEN')]) {
-                    sh "echo //registry.npmjs.org/:_authToken=$NPM_TOKEN > $WORKSPACE/.npmrc"
+                        sh "aws s3 cp ./dist/docs ${DOCS_CHANNEL_LOC} --recursive"
+                        sh "aws s3 cp ./dist/docs ${DOCS_VERSIONED_LOC} --recursive"
+                        sh "aws s3 cp ./dist/provider/app.json ${MANIFEST_LOC}"
+                    }
+                }
+                stage("Deploy Client to NPM (Production)") {
+                    input {
```

Comment:
Can you use `when` blocks on these?

I'm assuming the prompt would happen at this stage in the pipeline, after the tests have ran. If it's a trivial change, would be good to have the prompt right at the start of the build. Could move the `input` to the top of the script, with it's own `when` condition?..

Reply:
Ahhh yep, it'd be nice to set this as soon as you push versus having to wait around until the tests finish. Moving.